LoginAndAclSetup.md

```diff
@@ -0,0 +1,178 @@
+
+In order to use this script, an existing Data Lake Storage Gen2 storage account. Run `azd env set AZURE_ADLS_GEN2_STORAGE_ACCOUNT <your-storage-account>` prior to running the script.
+
+To run the script, run the following command: `./scripts/adlsgen2setup.ps1`. The script performs the following steps:
+* Creates 3 example [groups](https://learn.microsoft.com/azure/active-directory/fundamentals/how-to-manage-groups): `GPTKB_AdminTest`, `GPTKB_EmployeeTest`, `GPTKB_HRTest`
+* Creates a filesystem / container `gptkbcontainer` in the storage account.
+* Creates two directories, `benefitinfo` and `employeeinfo` in the `gptkbcontainer` filesystem / container.
+* Uploads the sample PDFs into both directories.
+* [Recursively sets Access Control Lists (ACLs)](https://learn.microsoft.com/azure/storage/blobs/data-lake-storage-acl-cli) on the `benefitinfo` and `employeeinfo` for the following groups:
+  * `GPTKB_AdminTest`: Can read all files in `gptkbcontainer`.
+  * `GPTKB_EmployeeTest`: Can only read files in `employeeinfo`.
+  * `GPTKB_HRTest`: Can read files in both `employeeinfo` and `benefitinfo`.
+
+In order to use the sample access control, you need to join these groups in your Azure AD tenant.
+
+Note that this optional scripts may not work in codespaces if your administrator has applied a [Conditional Access policy](https://learn.microsoft.com/azure/active-directory/conditional-access/overview) to your tenant.
+
+### Azure Data Lake Storage Gen2 Prep Docs
+
+Once a Data Lake Storage Gen2 storage account has been setup with sample data and access control lists, the [prepdocs.ps1](./scripts/prepdocs.ps1) can be used to automatically process PDFs in the storage account and store them with their [access control lists in the search index](https://learn.microsoft.com/azure/storage/blobs/data-lake-storage-access-control).
+
+To run this script with a Data Lake Storage Gen2 account, first set the following environment variables:
+
+* `AZURE_ADLS_GEN2_STORAGE_ACCOUNT`: Name of existing [Data Lake Storage Gen2 storage account](https://learn.microsoft.com/azure/storage/blobs/data-lake-storage-introduction).
+* (Optional) `AZURE_ADLS_GEN2_FILESYSTEM`: Name of existing Data Lake Storage Gen2 filesystem / container in the storage account. If empty, `gptkbcontainer` is used.
+* (Optional) `AZURE_ADLS_GEN2_FILESYSTEM_PATH`: Specific path in the Data Lake Storage Gen2 filesystem / container to process. Only PDFs contained in this path will be processed.
+
+Once the environment variables are set, run the script using the following command: `./scripts/prepdocs.ps1` or `./scripts/prepdocs.sh`.
+
+### Manually managing Document Level Access Control
+
+Manually enable document level access control on a search index and manually set access control values using the [manageacl.ps1](./scripts/manageacl.ps1) script.
```

Comment:
Is this an alternative to the data lake approach? It might be worth mentioning that at the beginning, just because it's easy to lose track of that in this long readme.

Reply:
Yes will do that